examples/hello-world/step-by-step/cifar10/sag_with_deploy_map/sag_deploy_map.ipynb

```diff
@@ -0,0 +1,282 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "514c47e2-420d-4af4-9bf0-cac337c51c39",
+   "metadata": {},
+   "source": [
+    "# FedAvg with SAG (Scatter & Gather) workflow with Site-Specific Configurations\n",
+    "\n",
+    "In this example, we will demonstrate the Feg Avg. SAG workflow using CIFAR10 dataset with site-specific configurations and deploy_map. \n",
+    "\n",
+    "We build on top of the previous example [FedAvg with SAG workflow](../sag/sag.ipynb#title)\n",
+    "\n",
+    "Different from the previous examples, we like to demonstrate how to deploy different \"app\" to different sites using different configurations. \n",
+    "\n",
+    "The next section is going to be the same as previous example, please follow the steps \n",
+    "  * [Understanding Fed Average and SAG](../sag/sag.ipynb#sag)\n",
+    "  * [Convert training code to federated learning training code](../sag/sag.ipynb#code)\n",
+    "  * [Prepare Data](../sag/sag.ipynb#data)\n",
+    "\n",
+    "\n",
+    "The next step is to preapre job folder and configurations. We will start from here. \n",
+    "\n",
+    "## Job Folder and Configurations\n",
+    "\n",
+    "Before we dive into the details of the job configurations, let's first introduce few terminologies.\n",
+    "\n",
+    "### Job and App\n",
+    "\n",
+    "So far, when we referring NVFLARE job, we are referring the job folder which consists \n",
+    "\n",
+    "  * The job folder structure\n",
+    "  * The job configuration and \n",
+    "  * The custom codes \n",
+    "\n",
+    "Actually, the complete picture of the job is the followings:\n",
+    "\n",
+    "  * The job has special folder structure\n",
+    "  * The job is consists of many \"apps\" or \"applications\"\n",
+    "  * each application is consists app's custom code and configurations. \n",
+    "  \n",
+    "  you probably has noticed the folder structure as \n",
+    "```\n",
+    "  <job folder>\n",
+    "  | -- app\n",
+    "  | -----config/\n",
+    "  | --------config_fed_client.conf\n",
+    "  | --------config_fed_server.conf\n",
+    "  | -----custom\n",
+    "  | --------train.py\n",
+    "    \n",
+    "```\n",
+    "  In such case, we only defined one app. We could define many apps. \n",
+    "  \n",
+    "  ```\n",
+    "  <job folder>\n",
+    "  | -- app_server\n",
+    "  | -- app_1\n",
+    "  | -- app_2\n",
+    "    \n",
+    "```\n",
+    "  in above,we defined three apps, where server, site-1, site-2 each has different app. \n",
```

Comment:
in above, we defined three apps: "app_server", "app_1" and "app_2".

Since we have not talked about meta.json here, these apps does not map to server, site-1, site-2 one to one.

We could have meta that contains a deploy map which maps app_1 -> site-2, and app_2 -> site-1, or app_1 -> site-1 and site-2